hydrolib/core/dflowfm/ext/models.py

```diff
@@ -181,6 +184,117 @@ def validate_location_type(cls, v: str) -> str:
+    """str: Space-uniform time series in <*.tim> file."""
+
+    allowedvaluestext = "Possible values: bcAscii, netcdf, uniform."
+
+
+class MeteoInterpolationMethod(str, Enum):
+    """
+    Enum class containing the valid values for the interpolationMethod
+    attribute in Meteo class.
+    """
+
+    nearestnb = "nearestNb"
+    """str: Nearest-neighbour interpolation, only with station-data in forcingFileType=netcdf"""
+
+    allowedvaluestext = "Possible values: nearestNb (only with station data in forcingFileType=netcdf ). "
+
+
+class Meteo(INIBasedModel):
+    """
+    A `[Meteo]` block for use inside an external forcings file,
+    i.e., a [ExtModel][hydrolib.core.dflowfm.ext.models.ExtModel].
+
+    All lowercased attributes match with the meteo input as described in
+    [UM Sec.C.5.2.3](https://content.oss.deltares.nl/delft3dfm1d2d/D-Flow_FM_User_Manual_1D2D.pdf#subsection.C.5.2.3).
+    """
+
+    class Comments(INIBasedModel.Comments):
+        quantity: Optional[str] = Field(
+            "Name of the quantity. See UM Section C.5.3", alias="quantity"
+        )
+        forcingfile: Optional[str] = Field(
+            "Name of file containing the forcing for this meteo quantity.",
+            alias="forcingFile",
+        )
+        forcingfiletype: Optional[str] = Field(
+            "Type of forcingFile.", alias="forcingFileType"
+        )
+        targetmaskfile: Optional[str] = Field(
+            "Name of <*.pol> file to be used as mask. Grid parts inside any polygon will receive the meteo forcing.",
+            alias="targetMaskFile",
+        )
+        targetmaskinvert: Optional[str] = Field(
+            "Flag indicating whether the target mask should be inverted, i.e., outside of all polygons: no or yes.",
+            alias="targetMaskInvert",
+        )
+        interpolationmethod: Optional[str] = Field(
+            "Type of (spatial) interpolation.", alias="interpolationMethod"
+        )
+
+    comments: Comments = Comments()
+
+    _disk_only_file_model_should_not_be_none = (
+        validator_set_default_disk_only_file_model_when_none()
+    )
+
+    _header: Literal["Meteo"] = "Meteo"
+    quantity: str = Field(alias="quantity")
+    forcingfile: ForcingModel = Field(alias="forcingFile")
+    forcingfiletype: MeteoForcingFileType = Field(alias="forcingFileType")
+    targetmaskfile: Optional[PolyFile] = Field(None, alias="targetMaskFile")
+    targetmaskinvert: Optional[bool] = Field(None, alias="targetMaskInvert")
+    interpolationmethod: Optional[MeteoInterpolationMethod] = Field(
+        alias="interpolationMethod"
+    )
+
+    def is_intermediate_link(self) -> bool:
+        return True
+
+    forcingfiletype_validator = get_enum_validator(
+        "forcingfiletype", enum=MeteoForcingFileType
+    )
+    interpolationmethod_validator = get_enum_validator(
+        "interpolationmethod", enum=MeteoInterpolationMethod
+    )
+
+    @property
+    def forcing(self) -> Optional[ForcingBase]:
```

Comment:
We could remove this property as I assume it only works with a `ForcingModel`.
We can add it once it is a requested functionality and think about a way to do this.

Reply:
ok, done in newer commit.